File: app/src/main/java/com/example/focusstartsecondpart/features/events/domain/EventsInteractorImpl.java
package com.example.focusstartsecondpart.features.events.domain;

import com.example.focusstartsecondpart.features.events.domain.model.Event;

import java.util.List;

import io.reactivex.Single;

public class EventsInteractorImpl implements EventsInteractor {

    private EventsRepository eventsRepository;

    public EventsInteractorImpl(EventsRepository eventsRepository){
        this.eventsRepository = eventsRepository;
    }

    @Override
    public Single<List<Event>> loadEvents() {
        return eventsRepository.loadEvents();
    }
}

File: app/src/main/java/com/example/focusstartsecondpart/features/events/domain/EventsInteractorImpl.kt
package com.example.focusstartsecondpart.features.events.domain

import com.example.focusstartsecondpart.features.events.domain.model.Event
import io.reactivex.Observable

import io.reactivex.Single

class EventsInteractorImpl(private val eventsRepository: EventsRepository) : EventsInteractor {
    override fun loadEvents(): Single<List<Event>> {
        return eventsRepository.loadEvents()
                .toObservable()
                .flatMap { listEvent: List<Event> -> Observable.fromIterable(listEvent) }
                .filter { event: Event -> event.status != -2 }
                .toList()
    }
}
